### ISSUE TITLE
Emojis/Icons for programming languages

### ISSUE THREAD

**Issue by christ-offer** [silverbulletmd/silverbullet#125]
Recently started using this for making personal knowledge bases, and although not a big thing, it would be nice to have emoji/icons/stickers for different programming languages. Any possibility of this? :)

**Comment by bbroeksema** [comment]
It's most likely "as complex" as making some changes to this file: https://github.com/silverbulletmd/silverbullet/blob/main/plugs/emoji/emoji.json -> adding the emoji and setting a suitable tag to it.

**Comment by zefhemel** [comment]
I don't fully understand, have emojis for programming language? Like a "Java" emoji? If you can map an emoji to an _existing_ emoji (part of the emoji standard), like :python: to 🐍  or something, @bbroeksema 's suggestion would work. Custom emojis (as in: custom images) would be a different story. Also doable, but a different implementation approach required.

**Comment by christ-offer** [comment]
Yeah I meant more like being able to use custom images probably. For example do :typescript and get something like this: https://slackmojis.com/emojis/1383-typescript

**Comment by zefhemel** [comment]
So basically this would amount to custom emoji right? That may be nice to have.

**Comment by Maarrk** [comment]
We have custom emoji configuration, we have image attachments, I think if we had syntax to include an image inline and make its height equal to line height, this could be implemented nicely.

---

Side effect: inline images acould be nice for e.g. runes and other cool symbols not in Unicode and/or the font

**Comment by zefhemel** [comment]
This is going to be my solution for everything now, but once [Space Lua](https://silverbullet.md/Space%20Lua) stabilizes I think we can indeed implement this feature in a library combining some images with a simple Lua function that renders as an image (I still have to decide if supporting literal HTML can be done somewhat securely). In that case the flow would be: install some Library that imports a bunch of programming language libraries, then you could inline their icons using e.g. `${lang "python"}` (which would then render as the Python logo.

**Comment by Maarrk** [comment]
I'd say that supporting `|inline` size specifier is easier to understand and doesn't need any library script. Both give it a lower barrier to entry, but I will have a more informed opinion when I actually attempt to write it and use a bit

**Comment by Maarrk** [comment]
It definitely needs some work in same places as #827 and #833, but is really simple: https://github.com/Maarrk/silverbullet/commit/62bcdaacbf3b9da4167789b7db67923d1b4b4fb6

![obraz](https://github.com/user-attachments/assets/070f41b9-3e40-43bd-992c-915f9a0d9671)

On the other hand, I'm not sure if it's worth pursuing this way at all, since some of the coolest places where we use emoji can't work with widgets:

![obraz](https://github.com/user-attachments/assets/86b202ad-d5e7-4de0-bdd5-3f5bb9849afd)

---

A much simpler alternative would be to write a nice documentation on how to serve more fonts that include the desired icons, and setup emoji aliases for these.

**Comment by zefhemel** [comment]
Great :)
